java/Oleksii-Riabov/src/com/zoolatech/lecture2/tasks/_2/Controller.java

```diff
@@ -0,0 +1,172 @@
+                    continue;
+                }
+            } else {
+                System.out.println("Enter integer value");
+                continue;
+            }
+        }
+        System.out.println("You create Room of " + (robot.getRoom().width() + 1) + " titles height and of " +
+                "" + (robot.getRoom().height() + 1) + " titles wide (W-H(0-" + robot.getRoom().width() + ",0-"
+                + robot.getRoom().height() + "))," + "\nRobot's direction is: "
+                + robot.getDirection() + ", \nRobot's coordinates are: " +
+                "Width-Height(" + robot.getWidth() + "," + robot.getHeight() + ").\n");
+    }
+
+    public void actions() {
+        Scanner input = new Scanner(System.in);
+
+        outer:
+        while (true) {
+            instructions();
+            System.out.println("Enter action");
+
+            String line = input.nextLine().toUpperCase();
+            switch (line) {
+                case "M" -> {
+                    System.out.println("Enter number of titles to move as integer value.");
+                    while (true) {
+                        Scanner scanner = new Scanner(System.in);
+                        int titles;
+                        if (scanner.hasNextInt()) {
+                            titles = scanner.nextInt();
+                            if (titles == 1) {
+                                robot.move();
+                                coordinates();
+                                break;
+                            } else {
+                                robot.move(titles);
+                                coordinates();
+                                break;
+                            }
+                        } else {
+                            System.out.println("Enter integer value");
+                            continue;
+                        }
+                    }
+                }
+                case "L" -> {
+                    robot.turnLeft();
+                    coordinates();
+                }
+                case "R" -> {
+                    robot.turnRight();
+                    coordinates();
+                }
+                case "C" -> {
+                    robot.getCoordinates();
+                    System.out.println();
+                }
+                case "D" -> {
+                    System.out.println("Robot's direction is: " + robot.getDirection());
+                    System.out.println();
+                }
+                case "I" -> instructions();
+                case "EXIT" -> {
+                    System.out.println("Program ends");
+                    break outer;
+                }
+                default -> System.out.println("Please enter valid value\n");
+            }
+        }
+    }
+
+    public void coordinates() {
+        System.out.println("Robot's direction is: " + robot.getDirection() + "\nRoom's size is:" +
```

Comment:
same - you could just use an overridden `toString`

Reply:
Also move this in "toString" method.